libbeat/publisher/pipeline/output.go

```diff
@@ -69,12 +71,18 @@ func (w *netClientWorker) run() {
 				return
 			}
 
+			if attemptReconnect {
+				logp.Info("Attempting to reconnect.")
+			}
```

Comment:
+1 for logging connection and reconnect attempts

We have two loops in this methods. First loop attempts to connect. Second loop processes events.

The first loop becomes active after beats startup, once the first event is to be published. That is connecting the outputs is done lazily. In this case the log message should indicate this being the initial connect.

Once the send loop fails, we're about to reconnect.

Having a counter on failed connect attempts would be nice to have as well.

E.g. for the lifetime of an output it would be nice to have log messages like (assuming we would have an 'identifier' per output): 

```
Startup output to <identifier>
Connect to <identifier>
Failed to connect to <identifier>: <reason>
Connect to <identifier>
Connection to <identifier> established
Failed to publish events to <identifier>: %v
Closed connection to <identifier>
Attempting to reconnect to <identifier> with 0 reconnect attempt(s).
Failed to connect to <identifier>: <reason>
Attempting to reconnect to <identifier> with 1 reconnect attempt(s).
Connection to <identifier> established
Shutdown output to <identifier>
```

That is, we have 4 phases/states to distinguish:
1. init worker
2. create initial connection
3. reconnect
4. shutdown

Having an enum to differentiate on state and logging should help here.

Reply:
I wanted to use hostname as an identifier. However, we don't have it here.
I'll be adding additional logging lines for other events.